pkg/dao/snapshots_test.go

```diff
@@ -86,9 +87,14 @@ func (s *SnapshotsSuite) TestCreateAndList() {
 
 	mockPulpClient := pulp_client.NewMockPulpClient(t)
 	sDao := snapshotDaoImpl{db: tx, pulpClient: mockPulpClient}
-	mockPulpClient.On("GetContentPath").Return(testContentPath, nil)
+	sDao.WithContext(context.Background())
 
-	repoDao := repositoryConfigDaoImpl{db: tx, yumRepo: &mockExt.YumRepositoryMock{}}
+	mockPulpClient.On("WithContext", context.Background()).Return(mockPulpClient).Twice()
+	mockPulpClient.On("GetContentPath").Return(testContentPath, nil).Twice()
+	mockPulpClient.On("WithDomain", "").Return(mockPulpClient)
```

Comment:
I think this PR is a step in the right direction, although it feels like it adds a lot more boiler plate (like this) to the tests which is not ideal.

Reply:
here's an idea (if you think it makes sense) .   We could add some methods to the mock that do this for us.  I imagine something like:

```
mockPulpClient.WithContext().WithDomain().
     On("GetContentPath").Return(testContentPath, nil).Twice()

```


and under the hood, this would mock WithContext & WithDomain  MinTimes(1)   so we'd lose the count on those particular mock calls, but I'm not sure we actually care for that to be precise.  I would imagine these methods would have to live in some new file like "pulp_client_mock_helpers.go"  or something?

Reply:
actually i realized MinTimes() is in go mock and not testify.  It looks like if you just don't specify Times()  it doesn't care the number

Reply:
I like this idea. I'll make that change.